### ISSUE TITLE
Cancel process for State internships

### ISSUE THREAD

**Issue by kbwagner** [openopps/OpenOppsTasks#207]
Needed by 10/29

State has requested the ability to cancel internships after previously saying they didn't need it.

Open Opps has existing cancel functionality we could consider bringing over.

Need to determine what would happen to a cancel that has applicants.

**Comment by abudz19** [comment]
@kbwagner Is there a good time to meet about this to get context and a deadline?

**Comment by kbwagner** [comment]
@abudz19 I'll be in the office tomorrow if you have time.  i think this will end up going into release 8.3 which is sometime in November so requirements this month.

The cancel opportunity process exists in Open Opps now.  State said they didn't need it.  But now they do.  So just need to figure out how it will work at each step.

**Comment by kbwagner** [comment]
@abudz19 thanks for your thoughts.  
The current cancel link is on the opportunity itself vs on the admin dashboard.  I think that still works here and I guess technically if the administrators clicked in they could also cancel.  We just need to track the cancel.
- I'll have to verify if there is a modal that makes them pause like we do with a delete - can verify that tomorrow
- so yeh this will add a few email notifications so will add Debbie to the convo on this one.  @dalenard21 

Thanks!

**Comment by dalenard21** [comment]
@kbwagner @abudz19 

There is a modal that pops up - tested it in UAT. 
<img width="913" alt="Cancel-opportunity" src="https://user-images.githubusercontent.com/17745649/66593941-fec41b80-eb65-11e9-8331-994cdddffe98.png">

Probably need to revise the content.

But, I think using the existing process is good.  Let's list out the emails we need and we can create a new issue for that?  Or add it here.

**Comment by abudz19** [comment]
Sounds good to me. Can you think of any besides the three I've listed above? @dalenard21, would you prefer them in a different issue?

**Comment by dalenard21** [comment]
@abudz19 @kbwagner let's keep the emails in this issue.

Description: create emails for when an internship is canceled. There are 3 scenarios, so 3 different email notifications to the student. Create email for the following scenarios:

- [ ] if they cancel before application deadline inform student that one of their selections has been canceled, they can continue with two or add a third.
- [ ] if they cancel during the selection process inform student that one of their selections has been canceled, and that if they are not selected for their first two preferences they will be considered for any open internships that are available (if they selected that option during the application process)
- [ ] if they cancel after selection inform the student that their internship was canceled. Its unfortunate, but it happens.

Emails will come from Open Opps/DOS.  

Audience: Students

Date needed: 8.3 - sometime in November, so need this by end of month.

POC: Katrina

**Comment by kbwagner** [comment]
@abudz19 the current Cancel this opportunity link resides in the right rail of the opportunity page.  Where do you suggest it go for the internship since they don't have a right rail?

<img width="383" alt="Screen Shot 2019-10-17 at 10 24 09 AM" src="https://user-images.githubusercontent.com/18709918/67018012-9676bc00-f0c8-11e9-8f4c-6dc67eecbba1.png">

**Comment by kbwagner** [comment]
not easily since i'm not on my work computer but here's a link to one https://openopps.uat.usajobs.gov/internships/1051

**Comment by kbwagner** [comment]
I'm checking with Brody on when it should be too late to cancel.  I think if here are applicants it's fine...but I'm thinking that once it gets to ATP it may be too much of a mess.  Which might mean only one email.  Stay tuned

**Comment by kbwagner** [comment]
@abudz19 the only thing is doesn't that section not always show?  I can just say on the bottom of the page but if that section is displaying to display to the right of Close internship as a link?

**Comment by abudz19** [comment]
I thought that Brice and I worked through that section as showing who applied before the opportunity closed. I remember discussing with him various ways it would be awkward if there were too many applicants.

**Comment by kbwagner** [comment]
Only community managers can see the applicants though.  So a creator wouldn't see that section at all.

**Comment by kbwagner** [comment]
Confirmed with Brody it's ok to cancel at anytime even if it's gone to ATP so I think the 3 points above are fine.

@dalenard21 we may try to get the emails into the sprint that starts Monday.  Can we work together on the email content next week?  I can try to draft up some bullet points at least.

**Comment by dalenard21** [comment]
Hi @kbwagner yes, I can work on these next week.

**Comment by dalenard21** [comment]
@kbwagner @abudz19 

Here's an email letting the student know that one of their selections has been canceled before application period closes.  

Email Subject - Department of State Student Internship Program (Unpaid) - One of your internship has been canceled.

Hello, [Username},
We’re writing to let you know that the Department of State has canceled one of your internship opportunity selections. 

You can select another internship to apply to, or you can continue with just the two internship you’ve already selected. 

If you’d like to add another internship, click the link below to make your selection before the application period closes.

[Sign into Open Opportunities to select another internship](link to Open Opps).

If you don’t want to select another internship, you don’t need to do anything. 

You can check the status of your application on your [Open Opportunities dashboard](link to Open Opps dashboard). If you need to make an update, don't forget to submit your application again.

Thank you,
The Open Opportunities Team 
U.S. Office of Personnel Management 
1900 E Street NW. 
Washington, DC 20415


Let me know what you think.

**Comment by dalenard21** [comment]
@kbwagner @abudz19 

Here's the second email for when DOS cancels an internship during the selection process.  I'm not sure about the (if you chose that option during the application process) - let me know if that's clear or if we need to say anything else.

Email Subject - Department of State Student Internship Program (Unpaid) - One of your internship selections has been canceled.

Hello, [Username}
We want to let you know that the Department of State has canceled one of your internship opportunity selections. 

If the Department of State does not select for your other internship selection(s), they will consider you for any open internships that are available (if you chose that option during the application process).

You can check the status of your application on your [Open Opportunities dashboard](link to Open Opps dashboard). If you need to make an update, don't forget to submit your application again.

Thank you,
 
The Open Opportunities Team 
U.S. Office of Personnel Management 
1900 E Street NW. 
Washington, DC 20415

**Comment by dalenard21** [comment]
@kbwagner @abudz19 here's the third email about internship being canceled.  Not sure what else we can say.  Feel free to edit.

**Email Subject** - Department of State Student Internship Program (Unpaid) - Your internship has been canceled.

Hello [Username],

We want to let you know that the Department of State has canceled your internship opportunity. 
If you have questions, please contact studentinternship@state.gov.
 
Thank you,
 
The Open Opportunities Team 
U.S. Office of Personnel Management 
1900 E Street NW. 
Washington, DC 20415

the studentinternship@state.gov should pop up an email.

**Comment by kbwagner** [comment]
@dalenard21 did you have any updates for the cancel confirm?  

Current screens shot:https://user-images.githubusercontent.com/17745649/66593941-fec41b80-eb65-11e9-8331-994cdddffe98.png

We just need to update the secondary text to something like "Are you sure you want to cancel this internship?"  "If you cancel this internship, the applicants will not be considered for this position."

**Comment by kbwagner** [comment]
I have written all of the tickets so once we confirm text for the alert this can be closed.  Thanks!

**Comment by dalenard21** [comment]
@kbwagner what about this?  I think this can cover the scenario of applicants who selected it as an option and those that haven't applied yet.

Are you sure you want to cancel this internship opportunity?

If you cancel this internship, applicants will no longer be able to apply to it.

**Comment by dalenard21** [comment]
@kbwagner how about this.  It's a bit wordy, so feel free to edit.

If you cancel this internship, applicants will no longer be able to apply to it. You will also remove any applicants who have already applied for this internship.

**Comment by kbwagner** [comment]
@dalenard21 it works - thanks.  It kind of has to be wordy!
